10.1.0/startup/generateStartupCommand.js

```diff
@@ -1,78 +1,180 @@
+function getStartupCommand(CMDFILE) {
+    const CUSTOM_STARTUP_CMD_FLAG = "/opt/startup/CUSTOM_STARTUP_CMD_FLAG";
+    fs.writeFileSync(CUSTOM_STARTUP_CMD_FLAG, "FALSE");
+    const PACKAGE_JSON_FLAG = "/opt/startup/PACKAGE_JSON_FLAG";
+    fs.writeFileSync(PACKAGE_JSON_FLAG, "FALSE");
+    const PROCESS_JSON_FLAG = "/opt/startup/PROCESS_JSON_FLAG";
+    fs.writeFileSync(PROCESS_JSON_FLAG, "FALSE");
 
-const CUSTOM_STARTUP_CMD_FLAG = "/opt/startup/CUSTOM_STARTUP_CMD_FLAG";
-fs.writeFileSync(CUSTOM_STARTUP_CMD_FLAG, "FALSE");
-if (startupCommand) {
-    fs.writeFileSync(CUSTOM_STARTUP_CMD_FLAG, "TRUE"); // set CUSTOM_STARTUP_CMD_FLAG for remote debugging
-}
+    const DEFAULTAPP = "/opt/startup/default-static-site.js";
+    var userStartupCommand = fs.readFileSync(CMDFILE, 'utf8').trim();
 
-// No user-provided startup command, check for scripts.start
-const PACKAGE_JSON_FLAG = "/opt/startup/PACKAGE_JSON_FLAG";
-fs.writeFileSync(PACKAGE_JSON_FLAG, "FALSE");
-if (!startupCommand) {
-    var packageJsonPath = "/home/site/wwwroot/package.json";
-    var json = fs.existsSync(packageJsonPath) && JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'))
-    if (typeof json == 'object' && typeof json.scripts == 'object' && typeof json.scripts.start == 'string') {
-        console.log("Found scripts.start in package.json")
-        startupCommand = 'npm start';
-        fs.writeFileSync(PACKAGE_JSON_FLAG, "TRUE"); // set PACKAGE_JSON_FLAG for remote debugging
+    if (userStartupCommand) {
+        fs.writeFileSync(CUSTOM_STARTUP_CMD_FLAG, "TRUE"); // set CUSTOM_STARTUP_CMD_FLAG for remote debugging
+    }
+    
+    // contains spaces, so multipart command
+    // eg: npm start
+    if (userStartupCommand.indexOf(" ") != -1) {
+        console.log("Found command: "+ userStartupCommand);
+        return userStartupCommand;
+    }
+
+    // look for package.json
+    if (userStartupCommand.endsWith("package.json") || !userStartupCommand) {
+        var packageJsonPath = "";
+        if (!userStartupCommand) {
+            packageJsonPath = "/home/site/wwwroot/package.json";
+        } else {
+            packageJsonPath = userStartupCommand;
+        }
+        if (fs.existsSync(packageJsonPath)) {
+            var json =  JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'))
+            if (typeof json == 'object' && typeof json.scripts == 'object' && typeof json.scripts.start == 'string') {
+                console.log("Found scripts.start in " + packageJsonPath);
+                fs.writeFileSync(PACKAGE_JSON_FLAG, "TRUE"); // set PACKAGE_JSON_FLAG for remote debugging
+                if (packageJsonPath != "package.json") {
+                    var packageJsonDir = packageJsonPath.substring(0, packageJsonPath.indexOf("package.json")-1);
+                    return "npm --prefix=" + packageJsonDir + " start";
+                } else {
+                    return 'npm start';
+                }
+            }
+        }
+    }
```

Comment:
if package.json is found, run with npm start
node wrapper will add inspect argument